cliconfig/config.go

```diff
@@ -0,0 +1,215 @@
+package cliconfig
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/pkg/homedir"
+)
+
+const (
+	// Where we store the config file
+	CONFIGFILE     = "config.json"
+	OLD_CONFIGFILE = ".dockercfg"
+	INDEXSERVER    = "https://index.docker.io/v1/"
```

Comment:
I would prefer to keep these variables inside the registry package.  They will likely disappear with the registry refactor, which I don't think should require changes here. Same goes for the functions which use these `IndexServerAddress()` and `IndexServerName()`